### PR TITLE
Add custom-server-express example

### PR DESCRIPTION
**Pull request by jaredpalmer**
I think the custom server example is fine, but it still took me a several minutes to setup express. 

Adding this should help clarify a common use case and more importantly, demonstrate that migrating to next isn't too tough.

### CODE REVIEW

File: examples/custom-server-express/server.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  server.use((req, res, next) => {
+    const accept = accepts(req)
+    const type = accept.type(['json', 'html'])
+    if (type === 'json') {
+      return handle(req, res)
+    }
+    next()
+  })
+
+  server.get('/a', (req, res, next) => {
```
Comment (arunoda) [inline examples/custom-server-express/server.js]: You don't need `next` argument here and the following cases.

**Comment by rauchg** [comment]
This is so beautiful @jaredpalmer

File: examples/custom-server-express/server.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+const accepts = require('accepts')
+
+const app = next('.', { dev: true })
+const handle = app.getRequestHandler()
+
+app.prepare()
+.then(() => {
+  const server = express()
+
+  server.use((req, res, next) => {
```
Comment (arunoda) [inline examples/custom-server-express/server.js]: @rauchg I wish we could have remove this by handling accept types inside `app.render`.
Then we don't need any `server.use` at all.

**Comment by jaredpalmer** [comment]
@rauchg thanks man. I appreciate all the work you and the zeit team do.

File: examples/custom-server-express/server.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+const express = require('express')
+const next = require('next')
+const accepts = require('accepts')
+
+const app = next('.', { dev: true })
```
Comment (nkzawa) [inline examples/custom-server-express/server.js]: The first argument is also optional and `.` is default, so it should be

```
next({ dev: true })
```

Comment (frol) [inline examples/custom-server-express/server.js]: I believe that explicit is better than implicit; please, don't hide this feature in the example.

Comment (jaredpalmer) [inline examples/custom-server-express/server.js]: @nkzawa whats the verdict?

Comment (thisguychris) [inline examples/custom-server-express/server.js]: how about adding a comment in the example stating `The first argument is also optional and . is default`

File: examples/custom-server-express/server.js (hunk 2 above)
Comment (rauchg) [inline examples/custom-server-express/server.js]: @nkzawa why did we not handle json directly in the render helper?

Comment (nkzawa) [inline examples/custom-server-express/server.js]: Because we want JSON endpoints to be fixed URL for component caching.

Comment (rauchg) [inline examples/custom-server-express/server.js]: Briliant

**Comment by jaredpalmer** [comment]
What's the status of this?

**Comment by nkzawa** [comment]
@jaredpalmer merged, thanks!

**Pull request merged**